server/src/main/java/com/google/coffeehouse/storagehandler/StorageHandler.java

```diff
@@ -0,0 +1,304 @@
+      }
+      if (!row.isNull(/*index=*/1) || !row.getString(/*index=*/1).isEmpty()) {
+        isbn = row.getString(/*index=*/1);
+      }
+      title = row.getString(/*index=*/2);
+    } else {
+      throw new IllegalArgumentException(BOOK_DOES_NOT_EXIST);
+    }
+    Book.Builder bookBuilder = Book.newBuilder(title);
+    // TODO: implement setting the bookId field @JosephBushagour
+    if (!author.isEmpty()) {
+      bookBuilder.setAuthor(author);
+    }
+    if (!isbn.isEmpty()) {
+      bookBuilder.setIsbn(isbn);
+    }
+    return bookBuilder.build();
+  }
+
+  /**
+  * Creates and returns a {@link Club} with the result of a strong read from the database.
+  * This method builds a {@link Club} with a club's ID, book ID, description, name, and ownerID.
+  *
+  * @param  dbClient  the database client
+  * @param  clubId    the club ID string used to query and get a club's information
+  * @return club      the Club object built containing the club information
+  */
+  public static Club getClub(DatabaseClient dbClient, String clubId) {
+    String bookId = "";
+    String description = "";
+    String ownerId = "";
+    String name = "";
+    // long resultCount = StorageHandlerHelper.getClubCount(dbClient, clubId);
+    // if (resultCount == 0) {
+    //   throw new IllegalArgumentException(CLUB_DOES_NOT_EXIST);
+    // }
+    Struct row = 
+      dbClient
+          .singleUse()
+          .readRow(
+            "Books",
+            Key.of(bookId),
+            Arrays.asList("bookId", "description", "name", "ownerId"));
+    if (row != null) {
+      bookId = row.getString(/*index=*/0);
+      description = row.getString(/*index=*/1);
+      name = row.getString(/*index=*/2);
+      ownerId = row.getString(/*index=*/3);
+    } else {
+      throw new IllegalArgumentException(CLUB_DOES_NOT_EXIST);
+    }
+    Book book = getBook(dbClient, bookId);
+    Club.Builder clubBuilder = Club.newBuilder(name, book);
+    // TODO: implement setting the clubId field @JosephBushagour
+    // TODO: implement setting the ownerId field @JosephBushagour
+    clubBuilder.setDescription(description);
+    return clubBuilder.build();
+  }
+
+  /**
+  * Creates and returns a list of {@link Persons}s that are a member of a club.
+  * This method builds a {@link Person} for each person who is a member of the club
+  * specified by club ID. Each {@link Person} is added to a list that gets returned.
+  *
+  * @param  dbClient    the database client
+  * @param  clubId      the club ID string used to query
+  * @return persons     the list of Person objects
+  */
+  public static List<Person> getListOfMembers(DatabaseClient dbClient, String clubId) {
+    List<Person> persons = new ArrayList<>();
+    long resultCount = StorageHandlerHelper.getMemberCount(dbClient, clubId);
+    if (resultCount < 0) {
+      throw new IllegalArgumentException(MembershipConstants.NO_MEMBERS);
+    }
+    Statement statement = 
+        Statement.newBuilder(
```

Comment:
I will be refactoring this code to also perform a strong read. I will be using read, which "reads zero or more rows from a database," within a transaction where I call `getMemberCount`. As far as I can tell in the documentation, there is no `readRows` since `read` is used for multiple rows.

Reply:
I was thinking of refactoring this code to also perform a strong read within a transaction, however I'm not sure if it is possible because the KeySet has to be specified, and I am unsure how to pass in club ID without needing to pass in user ID (since both are primary keys of the table).

Reply:
Actually, quick update on this. I think I can possibly use KeyRange? I'll get back to you on that if so.

Reply:
According to the [KeyRange documention](https://googleapis.dev/java/google-cloud-spanner/latest/com/google/cloud/spanner/KeyRange.html#prefix-com.google.cloud.spanner.Key-), I believe that [prefix](https://googleapis.dev/java/google-cloud-spanner/latest/com/google/cloud/spanner/KeyRange.html#prefix-com.google.cloud.spanner.Key-) allows for the specification of only one primary key